python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -599,6 +614,66 @@ def func(x):
 
         return self.agg(func)
 
+    def describe(self):
+        """
+        Generate descriptive statistics that summarizes the central tendency,
+        dispersion and shape of a dataset’s distribution, excluding NaN values.
+
+        Analyzes numeric DataFrames only
+
+        Parameters
+        ----------
+        DataFrame to be summarized
+
+        Returns
+        -------
+        DataFrame
+            Summary statistics of the Dataframe provided.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> gdf = cudf.DataFrame({"Speed": [380.0, 370.0, 24.0, 26.0],
+                                  "Score": [50, 30, 90, 80]})
+        >>> gdf
+        Speed  Score
+        0  380.0     50
+        1  370.0     30
+        2   24.0     90
+        3   26.0     80
+        >>> gdf.groupby('Score').describe()
+                count   mean   std    min    25%    50%    75%    max
+        Score
+        30         1  370.0  <NA>  370.0  370.0  370.0  370.0  370.0
+        50         1  380.0  <NA>  380.0  380.0  380.0  380.0  380.0
+        80         1   26.0  <NA>   26.0   26.0   26.0   26.0   26.0
+        90         1   24.0  <NA>   24.0   24.0   24.0   24.0   24.0
+
+        """
+        res = self.agg(
+            [
+                "count",
+                "mean",
+                "std",
+                "min",
+                _quantile_25,
+                _quantile_50,
+                _quantile_75,
+                "max",
+            ]
+        )
+        res.columns = [
+            "count",
+            "mean",
+            "std",
+            "min",
+            "25%",
+            "50%",
+            "75%",
+            "max",
+        ]
```

Comment:
This doesn't handle when there are multiple value columns in the groupby operation:
```python
>>> gdf = cudf.DataFrame({"Speed": [380.0, 370.0, 24.0, 26.0],
                                  "Score": [50, 30, 90, 80], "Other": [10, 20, 30, 40]})
>>> gdf.groupby('Score').describe()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/raid/wangm/dev/rapids/cudf/python/cudf/cudf/core/groupby/groupby.py", line 673, in describe
    "max",
  File "/raid/wangm/dev/rapids/cudf/python/cudf/cudf/core/dataframe.py", line 664, in __setattr__
    super().__setattr__(key, col)
  File "/raid/wangm/dev/rapids/compose/etc/conda/cuda_11.2/envs/rapids/lib/python3.7/contextlib.py", line 74, in inner
    return func(*args, **kwds)
  File "/raid/wangm/dev/rapids/cudf/python/cudf/cudf/core/dataframe.py", line 2708, in columns
    f"Length mismatch: expected {len(self._data.names)} elements ,"
ValueError: Length mismatch: expected 16 elements ,got 8 elements
```

This also relates to the question you PM'ed me - Pandas differ the column descriptions by appending extra levels in column names:

```python
      Speed                                               Other                                        
      count   mean std    min    25%    50%    75%    max count  mean std   min   25%   50%   75%   max
Score                                                                                                  
30      1.0  370.0 NaN  370.0  370.0  370.0  370.0  370.0   1.0  20.0 NaN  20.0  20.0  20.0  20.0  20.0
50      1.0  380.0 NaN  380.0  380.0  380.0  380.0  380.0   1.0  10.0 NaN  10.0  10.0  10.0  10.0  10.0
80      1.0   26.0 NaN   26.0   26.0   26.0   26.0   26.0   1.0  40.0 NaN  40.0  40.0  40.0  40.0  40.0
90      1.0   24.0 NaN   24.0   24.0   24.0   24.0   24.0   1.0  30.0 NaN  30.0  30.0  30.0  30.0  30.0
```